server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -362,24 +361,23 @@ private void updateContentRefs(Owner owner, Product product) {
     public Set<Content> getChangedContent(Owner owner, Collection<Content> content) {
         Set<Content> changed = Util.newSet();
 
-        log.debug("Syncing {} incoming content.", content.size());
+        log.debug("Syncing {} incoming content", content.size());
+        for (Content incoming : content) {
+            Content existing = this.contentCurator.lookupById(owner, incoming.getId());
```

Comment:
Instead of doing N lookups here, would it be better to somehow hit the DB once to get all the existing incoming?